fip-0017.md

```diff
@@ -5,246 +5,443 @@ status: Draft
 type: Functionality
 author: Pawel Mastalerz <pawel@dapix.io>
 created: 2020-09-17
-updated: 2020-09-17
+updated: 2020-10-23
 ---
 
 # Abstract
 Functionality proposed in this FIP is intended to make wrapping of FIO Tokens and Domains easier.
 
+Wrapping cannot be accomplished entirely inside the FIO Protocol and therefore will not be possible with this FIP alone. This FIP only defines the functionality required in FIO Protocol. For detailed overview of the greater initiative see [Enable FIO Token and Domain NFT wrapping](https://fioprotocol.atlassian.net/wiki/spaces/WP/pages/7012357/Enable+FIO+Token+and+Domain+NFT+wrapping).
+
 Proposed new actions:
-|Action|Endpoint|Description|
-|---|---|---|
-|wrapdomain|wrap_fio_domain|Transfers FIO Domain to designated FIO account.|
-|wraptoken|wrap_fio_token|Transfers FIO Token to designated FIO account.|
-||get_wrapped_fio_domains|Returns FIO Domains wrapped to account.|
-||get_wrapped_fio_tokens|Returns FIO Tokens wrapped to account.|
+|Contract|Action|Endpoint|Description|
+|---|---|---|---|
+|fio.wrap|wrapdomain|wrap_fio_domain|Transfers FIO Domain to designated FIO account on the way to another chain.|
```

Comment:
Let's go with something else to avoid confusion with eos.wrap which I think is the block producer approach for adjusting anything on chain.

We could use fio.wrapping. Not sure if there's a better concept than "wrap" since we're conceptually transferring things to other chains more than just wrapping them. I can't think of anything better at the moment that includes a positive and negative version of the same thing (wrap, unwrap), so I guess we go with this.

Reply:
Changed